Update upgrade guide for v10 (#5994)
cupy/cupy: docs/source/upgrade.rst

CUDA 10.1 or earlier is no longer supported.
Use CUDA 10.2 or later.

Dropping NCCL v2.4 / v2.6 / v2.7 Support
----------------------------------------

NCCL v2.4, v2.6, and v2.7 are no longer supported.

Dropping Python 3.6 Support
---------------------------

Python 3.6 is no longer supported.

Dropping NumPy 1.17 Support
---------------------------

> NumPy 1.17 is not longer supported.

Reviewer: ```suggestion
NumPy 1.17 is no longer supported.
```

> Changes in :class:`cupy.cuda.Stream` and :class:`cupy.cuda.Device` Behavior

Reviewer: Could you split the Device and Stream section?
Device things are more important for users (compared to Stream things) so I don't want to mix them.

```suggestion
Changes in :class:`cupy.cuda.Device` and :class:`cupy.cuda.Stream` Behavior
```

The old routines are deprecated.

> :func:`cupy.cuda.Device.use` now returns ``self`` to get aligned with :func:`cupy.cuda.Stream.use()`.

Reviewer: I think non-breaking changes need not be covered here. (otherwise, we have to cover most of the changes 😄 )

Author: Okay! I didn't think well that the change does not affect users' existing codes...

> :func:`cupy.ndarray.__pos__` behaves samely as :func:`cupy.positive` instead of returning ``self``.

Reviewer: ```suggestion
:func:`cupy.ndarray.__pos__` now returns the copy (samely as :func:`cupy.positive`) instead of returning ``self``.
```
